src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs

```diff
@@ -6,69 +6,81 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Configuration;
 using Microsoft.Extensions.Logging.Console;
+using Microsoft.Extensions.Options;
 using Serilog;
 using Serilog.Core;
+using Serilog.Extensions.Logging;
 using System;
 using System.Linq;
+using static Steeltoe.Extensions.Logging.DynamicLoggingBuilder;
 
 namespace Steeltoe.Extensions.Logging.DynamicSerilog
 {
     public static class SerilogLoggingBuilderExtensions
     {
+        public static ILoggingBuilder AddDynamicSerilog(this ILoggingBuilder builder, IConfiguration configuration, LoggerConfiguration serilogConfiguration = null, bool preserveDefaultConsole = false)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            // only run if an IDynamicLoggerProvider hasn't already been added
+            if (!builder.Services.Any(sd => sd.ServiceType == typeof(IDynamicLoggerProvider)))
+            {
+                if (!preserveDefaultConsole)
+                {
+                    builder.ClearProviders();
+                }
+
+                builder.AddFilter<SerilogDynamicProvider>(null, LogLevel.Trace); // TODO : What does this do?
+                builder.Services.AddOptions();
+
+                if (serilogConfiguration != null)
+                {
+                    builder.Services.Configure<SerilogOptions>(options => options.SetSerilogOptions(serilogConfiguration));
+                }
+                else
+                {
+                    builder.Services.Configure<SerilogOptions>(options => options.SetSerilogOptions(configuration));
```

Comment:
```suggestion
                    builder.Services
                        .AddOptions<SerilogOptions>()
                        .Configure<IConfiguration>((options, configuration) => options.SetSerilogOptions(configuration));
```
With this code we can avoid the usage of BuildServiceProvider to get the IConfiguration

Reply:
Thanks for suggestion, it is working well.